src/toil/fileStore.py

```diff
@@ -721,6 +721,13 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                                                               0.0, False)
         return localFilePath
 
+    def exportFile(self, jobStoreFileID, dstUrl):
+        while jobStoreFileID in self._pendingFileWrites:
```

Comment:
Can i pull a @hannes-ucsc on this and say that we have a CachingFileStore.harbingerFile.waitOnDownload method in place exactly for this reason?